itests/api_test.go

```diff
@@ -51,213 +48,142 @@ func (ts *apiSuite) testVersion(t *testing.T) {
 		lapi.RunningNodeType = lapi.NodeUnknown
 	})
 
-	ctx := context.Background()
-	apis, _ := ts.makeNodes(t, kit.OneFull, kit.OneMiner)
-	napi := apis[0]
+	full, _, _ := kit2.EnsembleMinimal(t, ts.opts...)
+
+	v, err := full.Version(context.Background())
+	require.NoError(t, err)
 
-	v, err := napi.Version(ctx)
-	if err != nil {
-		t.Fatal(err)
-	}
 	versions := strings.Split(v.Version, "+")
-	if len(versions) <= 0 {
-		t.Fatal("empty version")
-	}
+	require.NotZero(t, len(versions), "empty version")
 	require.Equal(t, versions[0], build.BuildVersion)
 }
 
-func (ts *apiSuite) testSearchMsg(t *testing.T) {
-	apis, miners := ts.makeNodes(t, kit.OneFull, kit.OneMiner)
-
-	api := apis[0]
-	ctx, cancel := context.WithCancel(context.Background())
-	defer cancel()
-	senderAddr, err := api.WalletDefaultAddress(ctx)
-	if err != nil {
-		t.Fatal(err)
-	}
-
-	msg := &types.Message{
-		From:  senderAddr,
-		To:    senderAddr,
-		Value: big.Zero(),
-	}
-	bm := kit.NewBlockMiner(t, miners[0])
-	bm.MineBlocks(ctx, 100*time.Millisecond)
-	defer bm.Stop()
-
-	sm, err := api.MpoolPushMessage(ctx, msg, nil)
-	if err != nil {
-		t.Fatal(err)
-	}
-	res, err := api.StateWaitMsg(ctx, sm.Cid(), 1, lapi.LookbackNoLimit, true)
-	if err != nil {
-		t.Fatal(err)
-	}
-	if res.Receipt.ExitCode != 0 {
-		t.Fatal("did not successfully send message")
-	}
-
-	searchRes, err := api.StateSearchMsg(ctx, types.EmptyTSK, sm.Cid(), lapi.LookbackNoLimit, true)
-	if err != nil {
-		t.Fatal(err)
-	}
-
-	if searchRes.TipSet != res.TipSet {
-		t.Fatalf("search ts: %s, different from wait ts: %s", searchRes.TipSet, res.TipSet)
-	}
-
-}
-
 func (ts *apiSuite) testID(t *testing.T) {
 	ctx := context.Background()
-	apis, _ := ts.makeNodes(t, kit.OneFull, kit.OneMiner)
-	api := apis[0]
 
-	id, err := api.ID(ctx)
+	full, _, _ := kit2.EnsembleMinimal(t, ts.opts...)
+
+	id, err := full.ID(ctx)
 	if err != nil {
 		t.Fatal(err)
 	}
-	assert.Regexp(t, "^12", id.Pretty())
+	require.Regexp(t, "^12", id.Pretty())
 }
 
 func (ts *apiSuite) testConnectTwo(t *testing.T) {
 	ctx := context.Background()
-	apis, _ := ts.makeNodes(t, kit.TwoFull, kit.OneMiner)
 
-	p, err := apis[0].NetPeers(ctx)
-	if err != nil {
-		t.Fatal(err)
-	}
-	if len(p) != 0 {
-		t.Error("Node 0 has a peer")
-	}
+	one, two, _, ens := kit2.EnsembleTwoOne(t, ts.opts...)
```

Comment:
nit: suggest renaming these ensemble methods to `EnsembleTwoFullOneMiner` etc for clarity

Reply:
Makes sense, but I think that's too long. We can find a convention like:

```go
kit2.Ensemble2M1F()
```

WDYT?

Reply:
I'd err on the side of being verbose - more of a nitpick though, we can address in a later PR